packages/server/src/routes.ts

```diff
@@ -30,7 +30,14 @@ export const createRouter = (ctx: AppContext): express.Router => {
       throw new ServerError(400, 'Invalid count parameter')
     }
     const count = Math.min(parsedCount, 1000)
-    const after = req.query.after ? new Date(req.query.after) : undefined
+    let after
+    if (req.query.after) {
+      const timestamp = Date.parse(req.query.after)
+      if (isNaN(timestamp)) {
+        throw new ServerError(400, 'Invalid date in after parameter')
+      }
+      after = new Date(timestamp)
+    }
```

Comment:
Nit: Could we keep the `const`, avoid a double call to `Date` functions, and make the change footprint smaller ?

```suggestion
    const after = req.query.after ? new Date(req.query.after) : undefined
    if (after != null && Number.isNaN(after.getTime())) {
      throw new ServerError(400, 'Invalid after parameter')
    }
```